website/docs/d/s3_bucket_objects.html.markdown

```diff
@@ -0,0 +1,51 @@
+---
+layout: "aws"
+page_title: "AWS: aws_s3_bucket_objects"
+sidebar_current: "docs-aws-datasource-s3-bucket-objects"
+description: |-
+    Returns keys and metadata of S3 objects
+---
+
+# Data Source: aws_s3_bucket_objects
+
+The bucket-objects data source returns keys (i.e., file names) and other metadata about objects in an S3 bucket.
+
+## Example Usage
+
+The following example retrieves a list of all object keys in an S3 bucket and creates corresponding Terraform object data sources:
+
+```hcl
+data "aws_s3_bucket_objects" "my_objects" {
+  bucket = "ourcorp"
+}
+
+data "aws_s3_bucket_object" "object_info" {
+  count  = "${length(data.aws_s3_bucket_objects.my_objects.keys)}"
+  key    = "${element(data.aws_s3_bucket_objects.my_objects.keys, count.index)}"
+  bucket = "${data.aws_s3_bucket_objects.my_objects.bucket}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `bucket` - (Required) Lists object keys in this S3 bucket
+* `prefix` - (Optional) Limits results to object keys with this prefix (Default: none)
+* `delimiter` - (Optional) A character used to group keys (Default: none)
+* `encoding_type` - (Optional) Encodes keys using this method (Default: none; besides none, only "url" can be used)
+* `max_keys` - (Optional) Maximum object keys to return or `-1` to retrieve all keys (Default: 1000)
```

Comment:
Let's remove the -1 part for now, folks can set this value to any large integer. 👍 

```suggestion
* `max_keys` - (Optional) Maximum object keys to return (Default: 1000)
```